src/run-function.ts

```diff
@@ -28,19 +33,39 @@ export const RunFunction = async (
 
   // App has indicated there's an unrecoverable error with this function invocation
   if (error) {
-    await client.apiCall("functions.completeError", {
+    const errorPayload = {
       error,
       function_execution_id: functionExecutionId,
-    });
+    };
+    if (debugMode) {
+      hookCLI.log("functions.completeError request payload:", errorPayload);
+    }
+    const errorResp = await client.apiCall(
+      "functions.completeError",
+      errorPayload,
+    );
+    if (debugMode) {
+      hookCLI.log("functions.completeError response payload:", errorResp);
+    }
     return;
   }
 
   // App has indicated its function completed successfully
   if (completed) {
-    await client.apiCall("functions.completeSuccess", {
+    const successPayload = {
       outputs,
       function_execution_id: functionExecutionId,
-    });
+    };
+    if (debugMode) {
+      hookCLI.log("functions.completeSuccess request payload:", successPayload);
```

Comment:
It would look something like this right now:
```
functions.completeSuccess request payload: {
  outputs: {
    array_of_booleans: [ true, false ],
    array_of_integers: [ 6, 0, 4 ],
    array_of_numbers: [ 4, 1, 6 ],
    array_of_slack_channels: [ "C029YT5KEMB", "C02ARHSB996" ],
    array_of_slack_timestamps: [ 1684369217969 ],
    array_of_slack_user_ids: [ "U058B1W9ZBL" ],
    array_of_slack_usergroup_ids: [ "S123456" ],
    array_of_strings: [ "yo", "lo" ],
    bool: false,
    extendabletypedobject: { bool: true, int: 12, num: 14, str: "nope", something: "else" },
    int: 1337,
    num: 64,
    slack_blocks: { blocks: [ { type: "section", text: [Object] } ] },
    slack_channel: "C029YT5KEMB",
    slack_date: "2023-05-17",
    slack_interactivity: {
      interactivity_pointer: "heeheehee",
      interactor: { id: "U058B1W9ZBL", secret: "12345" }
    },
    slack_message_context: {
      message_ts: "1684368274.182069",
      user_id: "U058B1W9ZBL",
      channel_id: "C029YT5KEMB"
    },
    slack_message_ts: "1684368274.182069",
    slack_timestamp: 1684369217,
    slack_user_context: { id: "U058B1W9ZBL", secret: "secret!" },
    slack_user_id: "U058B1W9ZBL",
    slack_usergroup_id: "S123456",
    str: "batman",
    typedobject: { bool: false, int: 1337, num: 13, str: "robin" },
    untypedobject: { something: "in the way" }
  },
  function_execution_id: "Fx05884ZU33P"
}
```

Note that deeply nested objects get sort of half-summarized, e.g. `slack_blocks: { blocks: [ { type: "section", text: [Object] } ] },`

Otherwise, it seems that the message boundary protocol implementation does some manner of prettifying by default. I think what is happening is that [it is coercing the data to a string](https://github.com/slackapi/deno-slack-protocols/blob/main/src/mod.ts#L48), and as a part of that calls `toString()` on the provided request payload object, which then gets semi-stringified and indented (but also does the summarization of deeply-nested objects).

WDYT? Does that change anything? Or should we intentionally stringify here in addition?

Reply:
That's super helpful thanks!

Since there are multiple contexts that we'll encounter here it seems like there may be some cases where important data could get summarized that prevent us from debugging. I'm thinking we should intentionally stringify here rather than relying on the coercion from the protocol lib

Reply:
Here's what stringifying the request paylods now looks like: 
```
functions.completeSuccess request payload: {
  "outputs": {
    "array_of_booleans": [
      true,
      false
    ],
    "array_of_integers": [
      6,
      0,
      4
    ],
    "array_of_numbers": [
      4,
      1,
      6
    ],
    "array_of_slack_channels": [
      "C029YT5KEMB",
      "C02ARHSB996"
    ],
    "array_of_slack_timestamps": [
      1684438477802
    ],
    "array_of_slack_user_ids": [
      "U058B1W9ZBL"
    ],
    "array_of_slack_usergroup_ids": [
      "S123456"
    ],
    "array_of_strings": [
      "yo",
      "lo"
    ],
    "bool": false,
    "extendabletypedobject": {
      "bool": true,
      "int": 12,
      "num": 14,
      "str": "nope",
      "something": "else"
    },
    "int": 1337,
    "num": 64,
    "slack_blocks": {
      "blocks": [
        {
          "type": "section",
          "text": {
            "type": "mrkdwn",
            "text": "*Where should we order lunch from?* Poll by <fakeLink.toUser.com|Mark>"
          }
        }
      ]
    },
    "slack_channel": "C029YT5KEMB",
    "slack_date": "2023-05-17",
    "slack_interactivity": {
      "interactivity_pointer": "heeheehee",
      "interactor": {
        "id": "U058B1W9ZBL",
        "secret": "12345"
      }
    },
    "slack_message_context": {
      "message_ts": "1684368274.182069",
      "user_id": "U058B1W9ZBL",
      "channel_id": "C029YT5KEMB"
    },
    "slack_message_ts": "1684368274.182069",
    "slack_timestamp": 1684438477,
    "slack_user_context": {
      "id": "U058B1W9ZBL",
      "secret": "secret!"
    },
    "slack_user_id": "U058B1W9ZBL",
    "slack_usergroup_id": "S123456",
    "str": "batman",
    "typedobject": {
      "bool": false,
      "int": 1337,
      "num": 13,
      "str": "robin"
    },
    "untypedobject": {
      "something": "in the way"
    }
  },
  "function_execution_id": "Fx058G9S03BM"
}
```